src/Voron/Impl/Paging/CryptoPager.cs

```diff
@@ -15,6 +15,23 @@ namespace Voron.Impl.Paging
     public class CryptoTransactionState
     {
         public Dictionary<long, EncryptionBuffer> LoadedBuffers = new Dictionary<long, EncryptionBuffer>();
+        /// <summary>
+        /// Used for computing the total memory used by the transaction crypto buffers
+        /// </summary>
+        //TODO: we can optimize this by hiding the dictionary and computing the total size when we enter/remove a buffer
```

Comment:
That would probably be best, yes. This might be called a LOT

Reply:
Maybe something similar to `IScratchSpaceMonitor`:

https://github.com/ravendb/ravendb/blob/v4.2/src/Voron/Impl/Scratch/IScratchSpaceMonitor.cs